doc/design/upgrade.md

```diff
@@ -1,20 +1,37 @@
 # Upgrade
 
-After installed VCH, VIC admin would want to upgrade existing instances to latest version. This document describes the upgrade requirement, and design options.
+After installed VCH, VIC admin would want to upgrade existing instances to latest version, to get new features or to fix exposed security issues. This document describes the upgrade requirement, and design options.
 
 ## Requirements
 
 ### Scope
 
-While a new version is ready, new version's vic-machine upgrade can upgrade one running or powered off VCH appliance. The containers managed by this VCH will be left in existing version. To have a new version container, user need to create new container from upgraded VCH instance. If existing container is stopped and restarted, the old version should be kept.
+While a new version is ready, new version's vic-machine can upgrade one running or powered off VCH appliance.
 
-VIC version includes three parts, "release tag"-"build id"-"github commit hash".
+For VCH appliance, vic-machine upgrade will detect the version difference between old VCH and itself, find out the metadata change in those two versions, and then migrate metadata, replace iso files. If anything wrong happens in this process, revert back to old version and status. VCH will have downtime during upgrade, but should not be broken after this process.
 
-vic-machine upgrade will rely on build id change only. User need to provide newer binary with bigger build id, to upgrade existing VCH.
+The containers managed by this VCH will be left in existing version. If container is running, it will work well if it's not coupled with appliance through port-forwarding. But after VCH is upgraded, the container management might not be fully functional if there is configuration or communication level changes.
```

Comment:
The containers managed by this VCH will be left running the existing version. If the container is running, it will work well if it's not coupled with appliance through port-forwarding. But after VCH is upgraded, the container management might not be fully functional if there is configuration or communication level changes.